dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs

```diff
@@ -32,11 +34,20 @@ private void Initialize(GXService providerService)
 			_endpoint = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.URI_ENDPOINT);
 			_accessKey = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.ACCESS_KEY);
 
-			if (!string.IsNullOrEmpty(_endpoint)) { 
+			if (!string.IsNullOrEmpty(_endpoint)) {
 
+				if (!string.IsNullOrEmpty(_accessKey)) 
 				_client = new EventGridPublisherClient(
 					new Uri(_endpoint),
 					new AzureKeyCredential(_accessKey));
+				else
+				//Try authenticating using AD
+				{
+					logger.Debug("Authentication using Oauth 2.0.");
```

Comment:
Would is be posible to use GXLogging instead of the direct logger? This way, we can reuse the logic. For instance, GXLogging checks for enabled logs in each method.
The project already has a reference to GXEventRouter which references GxClasses.

Reply:
Ready. Thanks.